nifi-ignite-extensions/nifi-ignite-processors/src/test/java/pl/touk/nifi/processors/PutIgniteRecordTest.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.ignite.configuration.ClientConnectorConfiguration;
+import org.apache.nifi.csv.CSVReader;
+import org.apache.nifi.csv.CSVUtils;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.schema.access.SchemaAccessUtils;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import pl.touk.nifi.ignite.testutil.IgniteTestUtil;
+import pl.touk.nifi.ignite.testutil.PortFinder;
+
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class PutIgniteRecordTest {
+
+    private final Relationship success = new Relationship.Builder()
+            .name("success")
+            .build();
+
+    private final String personCsvSchema =
+            "{ \"type\": \"record\", \"name\": \"person\", \"fields\": " +
+                    "[ { \"name\": \"first_name\", \"type\": \"string\" }, { \"name\": \"last_name\", \"type\": \"string\" }, { \"name\": \"age\", \"type\": \"int\" } ] }";
+
+    private Ignite igniteServer;
+    private Connection conn;
+    private TestRunner runner;
+    private CSVReader csvReader;
+    private Map<String, String> flowFileAttributes = new HashMap<>();
+
+    @Before
+    public void before() throws InitializationException, IOException, SQLException {
+        setupIgnite();
+
+        flowFileAttributes.put("csv.delimiter", ";");
+        flowFileAttributes.put("csv.schema", personCsvSchema);
+
+        csvReader = new CSVReader();
+
+        runner = TestRunners.newTestRunner(PutIgniteRecord.class);
+
+        runner.addControllerService("csv-reader", csvReader);
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY.getName(), SchemaAccessUtils.SCHEMA_TEXT_PROPERTY.getValue());
+        runner.setProperty(csvReader, CSVUtils.VALUE_SEPARATOR.getName(), ";");
+        runner.setProperty(csvReader, CSVReader.CSV_PARSER.getName(), CSVReader.CSV_PARSER.getDefaultValue());
+        runner.setProperty(csvReader, CSVUtils.FIRST_LINE_IS_HEADER.getName(), "false");
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_TEXT.getName(), "${csv.schema}");
+        runner.enableControllerService(csvReader);
+
+        runner.setProperty(PutIgniteRecord.RECORD_READER.getName(), "csv-reader");
+        runner.setProperty(PutIgniteRecord.CACHE_NAME.getName(), "person");
+        runner.setProperty(PutIgniteRecord.CACHE_KEY_TYPE.getName(), "person_key");
+        runner.setProperty(PutIgniteRecord.CACHE_VALUE_TYPE.getName(), "person_value");
+        runner.setProperty(PutIgniteRecord.KEY_FIELD_NAMES.getName(), "first_name,last_name");
+    }
+
+    @After
+    public void after() throws SQLException {
+        conn.close();
+        igniteServer.close();
+    }
+
+    @Test
+    public void testProcessor() throws SQLException {
```

Comment:
What do you think about a couple more tests to better explain what's happening here? I feel like at least these two scenarios deserve to be covered:
1. More than one row in flow file
2. Overwriting of an existing cache key

Reply:
Ur right, it would be good to covert those cases.